### PR TITLE
Detect and report attempt to use abstract event through base

### LINKED ISSUE #406: Call to Abstract Event: result in BAD IL generation, instead of compilation error
Call to abstract event compiles (even in CTP5), results in BAD IL generation when trying to call it.

Sample code:
http://tryroslyn.azurewebsites.net/#b:releases-Dev14CTP5/K4Zwlgdg5gBAygTxAFwKYFsDcAobECG6qIADvgMaowDCA9hCLQDaoCCJJTY5+yY9AZmwBvbDBglgAIy7kY+KSgBOFZDHJN8IEDABCW1AHUwyABatFyFeWQBRAG6oIyMTFHjxkmd3mXra1EdnGFYbfggYAFkEBydkHHEAX1xPaVl1TW14YBJUJWMzAHlHJSUwABMnWOCALj0DAvM/VWqXcXcPL3TaErLKmEC4kLD6KJigto8Oj3ly8rcYKQMAOmjWmABqAF4Ye3wmYFRMGESYAHoz+AAJQoBVABkAET1bGABaD4A+OrylWiUaHAAAwAJiBAFY6tR8BAILQ1DwmEx5BEFMpVIsDDAiOgpHk6gByAAKfygKnQy30ICMJia6JsrVW4ziBNcMyUGB6VGEmOpTPWbx2ewORxO50ucBuD2euleHze3wGpX+gNBEKhMLhCP2yJhvnpaiW1OxGDxSkJJNoZMIlIatIsBsZawmrJmyTZEjSPnstAqMAAYrRaAAKACUHumHiNqH5E02O0DtGOF2udyeL3eXx+yoB1GBYMhNE18PUOpR+qsGOjJtx+JgxNJ5Nt1MaDsrDImsZZHuSSRSGS0Okt1vQrkjKF43t980i+EgYYjHvEcByeUaxV+FSqcZAAHdaFQdhBULvsrl8rSN6UtxBWmGEjMYHuD13gtsonOIA++33kkAAA==

### PR DESCRIPTION
**Pull request by gafter**
Fixes #406

### CODE REVIEW

**Comment by gafter** [comment]
@jaredpar @VSadov @AlekseyTs @agocke @VladimirReshetnikov Could a couple of you review this please?

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs [hunk 1]
```diff
@@ -1752,7 +1752,12 @@ private bool CheckEventValueKind(BoundEventAccess boundEvent, BindValueKind valu
                         // NOTE: availability of well-known members is checked in BindEventAssignment because
                         // we don't have the context to determine whether addition or subtraction is being performed.
 
-                        if (ReportUseSiteDiagnostics(eventSymbol, diagnostics, eventSyntax))
+                        if (receiver?.Kind == BoundKind.BaseReference && eventSymbol.IsAbstract)
+                        {
+                            Error(diagnostics, ErrorCode.ERR_AbstractBaseCall, boundEvent.Syntax, eventSymbol);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: It feels like we should detect this error in BindEventAccess.

Comment (agocke) [inline src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: Moreover, this code is duplicated in at least three places already. Is there some way to share this diagnostic across binding? This is a very general diagnostic.

Comment (agocke) [inline src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: I see -- we have no shared hierarchy between properties and events and this check is currently performed in CheckPropertyValueKind for properties.

In that case, this LGTM.

**Comment by agocke** [comment]
:+1:

File: src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs (hunk 1 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: @AlekseyTs This diagnostic is checked here by analogy with the code to check the corresponding conditions for properties.

**Pull request merged**